src/dmd/expressionsem.d

```diff
@@ -11486,6 +11486,10 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             {
                 return false;
             }
+
+            if (t1n.isfloating() && t2n.isfloating())
```

Comment:
I find this a bit arbitrary... shouldn't there be a test whether it's a scalar or not? Why is there no test needed here for `int` for example?

Reply:
because for integral types it already get optimized to use mem_cmp

for floating points it can't, and fallsback to using TypeInfo, wich is not available in betterC / wasm

https://github.com/dlang/dmd/blob/831d407006dcf82c452517f9446ece4d01621fab/src/dmd/e2ir.d#L1865